pyriemann/datasets/sampling.py

```diff
@@ -41,8 +42,66 @@ def _pdf_r(r, sigma):
     return np.exp(partial_1 + partial_2)
 
 
+def _slice_one_sample(ptarget, x0, w, rs):
+    """Slice sampling for one sample
+
+    Parameters
+    ----------
+    ptarget : function with one input
+        The target pdf to sample from or a multiple of it.
+    x0 : array
+        Initial state for the MCMC procedure. Note that the shape of this array
+        defines the dimensionality n_dim of the data points to be sampled.
+    w : float
+        Initial bracket width.
+    rs : int, RandomState instance or None
+        Pass an int for reproducible output across multiple function calls.
+
+    """
+    xt = np.copy(x0)
+    n_dim = len(x0)
+
+    for i in range(n_dim):
+
+        ei = np.zeros(n_dim)
+        ei[i] = 1
+
+        # step 1 : evaluate ptarget(xt)
+        Px = ptarget(xt)
+
+        # step 2 : draw vertical coordinate uprime ~ U(0, ptarget(xt))
+        uprime_i = Px * rs.rand()
+
+        # step 3 : create a horizontal interval (xl_i, xr_i) enclosing xt_i
+        r = rs.rand()
+        xl_i = xt[i] - r * w
+        xr_i = xt[i] + (1-r) * w
+        while ptarget(xt + (xl_i - xt[i]) * ei) > uprime_i:
+            xl_i = xl_i - w
+        while ptarget(xt + (xr_i - xt[i]) * ei) > uprime_i:
+            xr_i = xr_i + w
+
+        # step 4 : loop
+        while True:
+            xprime_i = xl_i + (xr_i - xl_i) * rs.rand()
+            Px = ptarget(xt + (xprime_i - xt[i]) * ei)
+            if Px > uprime_i:
+                break
+            else:
+                if xprime_i > xt[i]:
+                    xr_i = xprime_i
+                else:
+                    xl_i = xprime_i
+
+        # store coordinate i of new sample
+        xt = np.copy(xt)
+        xt[i] = xprime_i
+
+    return xt
+
+
 def _slice_sampling(ptarget, n_samples, x0, n_burnin=20, thin=10,
-                    random_state=None):
+                     random_state=None, n_jobs=1):
```

Comment:
To suppress flake8 error:
```suggestion
                    random_state=None, n_jobs=1):
```